data_structures/abstract_heap.py

```diff
@@ -28,6 +28,10 @@ def peek(self) -> T:
     def is_empty(self) -> bool:
         return len(self) == 0
 
+    @abstractmethod
+    def is_full(self) -> bool:
```

Comment:
I'd argue `is_full` doesn't belong to the abstract class as depending on the implementation, it may not have limited capacity.